### DISCUSSION TITLE
Specialized Groups

### DISCUSSION THREAD

**Discussion by tuxagon** [testdouble/double-up#64] [Ideas]
The goal of this change is to support having specialized groups that can be managed outside the `matchmaking.yml` file.

## Channel management

There should be a separate link in the navigation that takes you to a listing of the channels currently configured. The yaml file's should be listed there, but not be configurable from the UI.

Also on that page should be a create button that takes you to a form that has four fields:
* "Enabled" as a checkbox with a default of checked
* "Channel Name" to be the name of the channel in slack to use. Consider validating the channel exists before saving. No default.
* "Schedule" as a dropdown with daily, weekly, fortnightly, and monthly. No default.
* "Group Size" as a numerical field (or text box). Default to 2.

Saving should create a new `MatchmakingChannel` (or similar name).

On the listing page, all the details should be visible and there be a way to edit or disable any channels created through the UI.

Edit takes you to a new page that is like new but with filled in fields.

Disable stays on the page, but colors the channel differently.

The three states of  color on the listing should be
- Readonly
- Disabled
- Enabled

## Matchmaking

The initial group aggregation will need to fetch the groups and include them with the ones from the config file. It should handle the rest without issue since it simply loops through the list of active channels.

## Tagging

The idea of tagging is an extra special one because it could be a solution by itself, however, it comes with separate complexities. One of which is changing matchmaking in a way that keeps the scoring by takes into account another variable.

In essence, the idea behind tags is that a user can set words to be tied to their profile in Double Up. One of the channels would be configured to care about specific tags. The tags the channel cares about become qualifiers for a match to happen.

For example, if channel `#rotating-fun-time-club` looks for tags `gamers` and `jokesters`, then people in that channel with `gamers` would be matched together and `jokesters` would be matched together. There are some caveats with that, however.

1. If the matches are exclusive, then a person could be matched multiple times potentially. It's unlikely that person will want multiple matches. This could be solved by having a single tag for each channel, but that does become a more complicated UI concept for tracking.
2. If the tags are priorities, where the tags are matched with others with the same tag first and then it goes through matchmaking like normal, then multiple tags get confusing. For a single person, how do we track which is prioritized for matching?

Tagging reduces the number of channels needed potentially, but it is unclear whether that will be or is a problem.

**Comment by alimi** [comment]
Thanks for putting this together! I think it makes sense to start with channel management and pause on tagging until there's a bigger need for it.

In channel management, the three states are listed as Readonly, Disabled, and Enabled. What does Readonly do, and how is it different from Disabled?

**Reply by tuxagon** [reply]
I view "readonly" as the ones included from the config. We don't want those to be editable, at least at this stage, due to them being configured elsewhere. I thought it'd be useful to include the main config in the actual configured channel listing.

**Reply by alimi** [reply]
Ah right! Yep, that makes sense.
